src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,90 @@
+package duke.parser;
+
+import duke.command.Command;
+import duke.command.EnumCommand;
+import duke.command.ByeCommand;
+import duke.command.ListCommand;
+import duke.command.DoneCommand;
+import duke.command.TodoCommand;
+import duke.command.DeadlineCommand;
+import duke.command.EventCommand;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+
+import duke.exception.DukeException;
+
+/**
+ * A class to analyse the user's input and determine what command it is.
+ */
+public class Parser {
+    public Parser() {
+        
+    }
+
+    /**
+     * A function to parse the input from the user and turn it into a command.
+     * @param nextCommand the string of instruction input by the user
+     * @return Command that represents what the user wants Duke to do.
+     * @throws DukeException if the user's input is not a valid Command type for Duke.
+     */
+    public static Command parse(String nextCommand) throws DukeException {
+        String[] nextCommandArr = nextCommand.split(" ", 2);
+        Command next;
+        EnumCommand enumCommand;
+        
+        String command = nextCommandArr[0];
+        
+        if (command.equals("bye")) {
+            enumCommand = EnumCommand.BYE;
+        } else if (command.equals("list")) {
+            enumCommand = EnumCommand.LIST;
+        } else if (command.equals("done")) {
+            enumCommand = EnumCommand.DONE;
+        } else if (command.equals("todo")) {
+            enumCommand = EnumCommand.TODO;
+        } else if (command.equals("deadline")) {
+            enumCommand = EnumCommand.DEADLINE;
+        } else if (command.equals("event")) {
+            enumCommand = EnumCommand.EVENT;
+        } else if (command.equals("delete")) {
+            enumCommand = EnumCommand.DELETE;
+        } else if (command.equals("find")) {
+            enumCommand = EnumCommand.FIND;
+        } else {
+            throw new DukeException("Sorry, I don't know what that means~");
+        }
+        
+        switch (enumCommand) {
+            case BYE: 
+                next = new ByeCommand();
+                break;
+            case LIST: 
+                next = new ListCommand();
+                break;
+            case DONE: 
+                next = new DoneCommand(nextCommandArr);
+                break;
+            case TODO: 
+                next = new TodoCommand(nextCommandArr);
+                break;
+            case DEADLINE: 
+                next = new DeadlineCommand(nextCommandArr);
+                break;
+            case EVENT: 
+                next = new EventCommand(nextCommandArr);
+                break;
+            case DELETE: 
+                next = new DeleteCommand(nextCommandArr);
+                break;
+            case FIND:
+                next = new FindCommand(nextCommandArr);
+                break;
+            default:
+                throw new DukeException("Sorry, I don't know what that means~");
```

Comment:
Very nice and clear way of dealing with each type of command!!

But I think you don't have to have an indentation for your case statement, so case should have the same indentation as switch

Reply:
Yup! Thanks for pointing it out! I'm still not used to the coding convention for this mod hahaha